### PR TITLE
Changes to training via debate

### PR DESCRIPTION
**Pull request by david-lindner**
This adds a few things necessary for training MNIST classifiers via debate:
- parameter to control whether to update from won debates only
- evaluation of pre-computed debate results
- training from pre-computed debate results

### CODE REVIEW

**Comment by luuf** [comment]
How about you approve my PR and I look at yours

File: ai-safety-debate/precompute_debate_results.py [hunk 1]
```diff
@@ -62,9 +81,9 @@ def get_debate_results(start_point, batch_size, N_train, N_to_mask, judge_path):
 
     args = parser.parse_args()
 
-    results = np.zeros((args.N_train, 10))
-    batch_size = math.ceil(args.N_train / args.N_threads)
-    start_points = [i * batch_size for i in range(args.N_threads)]
+    results = np.zeros((args.N_train - args.start_sample, 10))
```
Comment (luuf) [inline ai-safety-debate/precompute_debate_results.py]: If I interpret this right, N_train isn't the Number of train samples you run anymore, but the index of the train sample you run up to. Maybe rename it to end_sample or train_until or something like that, instead?

File: ai-safety-debate/train_classifier_via_debate.py [hunk 2]
```diff
@@ -54,6 +56,18 @@ def run(
...
 
+    judge_accuracy = judge.evaluate_accuracy()
+    print("Judge accuracy:", judge_accuracy)
+
+    if precomputed_debate_results_path is not None:
+        debate_results = np.fromfile(precomputed_debate_results_path).reshape(
+            -1, 10, 10
+        )
+        print("Loaded debate results from", precomputed_debate_results_path)
+        print("These will be used for training instead of re-running the debates.")
```
Comment (luuf) [inline ai-safety-debate/train_classifier_via_debate.py:72]: Could be nice to assert that cheat_debate isn't true here

**Review by luuf — CHANGES_REQUESTED** [review]

File: ai-safety-debate/train_classifier_via_debate.py (hunk 2 above)
Comment (david-lindner) [reply ai-safety-debate/train_classifier_via_debate.py:72]: Agreed, will do!

File: ai-safety-debate/precompute_debate_results.py (hunk 1 above)
Comment (david-lindner) [reply ai-safety-debate/precompute_debate_results.py]: Yeah I wanted it to still be the number of samples, but it is just wrong :laughing: 
Will fix this and also add the option to precompute debates on a test set, which is what we probably want to evaluate.

**Comment by david-lindner** [comment]
Precomputation should be better now :laughing:

**Review by luuf — APPROVED** [review]

**Pull request merged**